neqo-transport/src/connection/mod.rs

```diff
@@ -1737,6 +1737,7 @@ impl Connection {
         // Handle each packet in the datagram.
         while !slc.is_empty() {
             self.stats.borrow_mut().packets_rx += 1;
+            self.stats.borrow_mut().dscp_rx[tos.into()] += 1;
```

Comment:
Moved from below. Tests rightfully assert `packets_rx == dscp_rx[Cs0]`.

Previously, if any of the packets would be padding, i.e. garbage below), the two stats would diverge, as the former is counted above, the latter is counted below.